datasets-sars-cov-2/0.6.3/Dockerfile

```diff
@@ -0,0 +1,87 @@
+# Use ubuntu as base image
+FROM ubuntu:focal as app
+
+ARG DATASETS_SARS_COV_2_VER=0.6.3
+
+# metadata
+LABEL base.image="ubuntu:focal"
+LABEL software="datasets-sars-cov-2"
+LABEL software.version="v0.6.3"
+LABEL dockerfile.version="1"
+LABEL description="Benchmark datasets for WGS analysis of SARS-CoV-2."
+LABEL website="https://github.com/CDCgov/datasets-sars-cov-2"
+LABEL license="https://github.com/CDCgov/datasets-sars-cov-2/blob/master/LICENSE"
+LABEL maintainer="Kelsey Florek"
+LABEL maintainer.email="kelsey.florek@slh.wisc.edu"
+
+# install dependencies
+ARG TZ=Etc/UTC
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y \
+ wget\
+ build-essential\
+ libxml2\
+ libmagic-dev\
+ libhdf5-dev\
+ git\
+ openjdk-8-jdk\
+ software-properties-common\
+ lsb-release\
+ apt-transport-https\
+ ca-certificates\
+ doxygen\
+ uuid-runtime && \
+ apt-get autoclean && \
+ rm -rf /var/lib/apt/lists/*
+
+# install latest cmake
+RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null && \
+ apt-add-repository "deb https://apt.kitware.com/ubuntu/ $(lsb_release -cs) main" && \
+ apt-get update && apt-get install -y cmake
+
+# get NCBI SRA Tools
+RUN mkdir /ncbi && cd /ncbi && \
+ git clone https://github.com/ncbi/sra-tools.git && \
+ git clone https://github.com/ncbi/ngs.git && \
+ git clone https://github.com/ncbi/ncbi-vdb.git
+
+RUN cd /ncbi/ncbi-vdb && ./configure && \
+ make && \
+ make install
+
+RUN cd /ncbi/ngs && ./configure && \
+ make -C ngs-sdk && \
+ make -C ngs-sdk install && \
+ make -C ngs-java && \
+ make -C ngs-java install && \
+ make -C ngs-python
+
+RUN cd /ncbi/sra-tools && ./configure && \
+ make && \
+ make install
+
+# install edirect
+RUN mkdir /edirect && \
+    perl -MNet::FTP -e '$ftp = new Net::FTP("ftp.ncbi.nlm.nih.gov", Passive => 1);$ftp->login; $ftp->binary;$ftp->get("/entrez/entrezdirect/edirect.tar.gz");' && \
+    gunzip -c edirect.tar.gz | tar xf - && \
+    rm edirect.tar.gz
+
+# vdb-config workaround: https://github.com/ncbi/sra-tools/issues/291
+RUN mkdir /home/user/ /home/user/.ncbi/ && printf '/LIBS/GUID = "%s"\n' `uuidgen` > /home/user/.ncbi/user-settings.mkfg
+
+# get datasets-sars-cov-2
+RUN cd /home/user/ && git clone --depth 1 --branch v${DATASETS_SARS_COV_2_VER} https://github.com/CDCgov/datasets-sars-cov-2.git
+
+ENV HOME /home/user/
+
+ENV PATH=$PATH:/edirect:/usr/local/ncbi/sra-tools/bin:/home/user/datasets-sars-cov-2/scripts
+
+WORKDIR /data
+
+# set perl locale settings
+ENV LC_ALL=C
+
+FROM app as test
+
+# download the smallest data set as a test
+RUN GenFSGopher.pl -o vocvoi-output /home/user/datasets-sars-cov-2/datasets/sars-cov-2-voivoc.tsv --numcpus $(nproc --all) --layout onedir --compressed
```

Comment:
Curious minds want to know when you installed perl into this container, because it's not listed.

Reply:
It's included in the base image `ubuntu:focal`

```
$ docker run --rm -ti ubuntu:focal
root@b8e95b51a5bf:/# perl --version

This is perl 5, version 30, subversion 0 (v5.30.0) built for x86_64-linux-gnu-thread-multi
(with 50 registered patches, see perl -V for more detail)

Copyright 1987-2019, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at http://www.perl.org/, the Perl Home Page.
```

Reply:
pretty sure `perl` is central/required for ubuntu, especially for `apt`